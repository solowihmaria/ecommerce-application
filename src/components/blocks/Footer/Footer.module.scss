@use '../../../assets/styles/constants.module' as c;

.footer {
    padding: 1rem 0 0;
    color: c.$color-white;
    background-color: c.$color-text-dark;
}

.container {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 100%;
    margin: 0 auto;
}

.columnsWrapper {
    display: flex;
    flex-wrap: wrap;
    gap: 2.5rem;
    justify-content: center;
    margin-bottom: 2rem;

    @media (width >= 680px) {
        gap: 1.5rem;
        justify-content: space-between;
        width: 90%;
    }

    @media (width >= 480px) {
        flex-direction: row;
    }
}

.columnTitle {
    margin-bottom: 1.25rem;
    font-size: 1.125rem;
    font-weight: 600;
    color: c.$color-white;
}

.linksList {
    display: flex;
    flex-direction: column;
    gap: 0.5rem;
    margin: 0;
    padding: 0;
    list-style: none;
}

.link {
    font-size: 0.9375rem;
    color: rgba(c.$color-white, 0.7);
    text-decoration: none;
    transition: color 0.2s ease;

    &:hover {
        color: c.$color-primary-light;
    }
}

.contactInfo {
    display: flex;
    flex-direction: column;
    gap: 0.75rem;
    margin-bottom: 1.25rem;
}

.contactItem {
    display: flex;
    gap: 0.75rem;
    align-items: center;
    font-size: 0.9375rem;
    color: rgba(c.$color-white, 0.7);
}

.contactIcon {
    min-width: 1.25rem;
    font-size: 1rem;
    color: c.$color-primary-light;
}

.workingHours {
    margin-bottom: 1.25rem;
    font-size: 0.8125rem;
    color: rgba(c.$color-white, 0.6);
}

.socialLinks {
    display: flex;
    gap: 0.75rem;
}

.socialButton {
    width: 2rem;
    height: 2rem;
    padding: 0;
    border-color: rgba(c.$color-white, 0.2);
    border-radius: 50%;
    color: rgba(c.$color-white, 0.7);

    &:hover {
        border-color: c.$color-primary-light;
        color: c.$color-white;
        background-color: transparent;
    }
}

.copyright {
    width: 100%;
    border-top: 1px solid rgba(c.$color-white, 0.1);
}

.copyrightContent {
    display: flex;
    flex-direction: column;
    gap: 0;
    align-items: center;

    @media (width >= 768px) {
        justify-content: space-between;
    }
}

.copyrightText {
    margin-bottom: 0;
    font-size: 0.875rem;
    color: rgba(c.$color-white, 0.6);
}

.disclaimerText {
    font-size: 0.6875rem;
    color: rgba(c.$color-white, 0.5);
    text-align: center;

    @media (width >= 768px) {
        text-align: right;
    }
}
