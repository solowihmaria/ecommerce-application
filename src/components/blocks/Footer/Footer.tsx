import { Button } from '../../ui/Button/Button';
import {
    FaMapMarkerAlt,
    FaPhone,
    FaEnvelope,
    FaInstagram,
    FaFacebook,
    FaTwitter,
} from 'react-icons/fa';
import { Link } from 'react-router-dom';
import styles from './Footer.module.scss';

export function Footer() {
    return (
        <footer className={styles.footer}>
            <div className={styles.container}>
                <div className={styles.columnsWrapper}>
                    <div className={styles.column}>
                        <h3 className={styles.columnTitle}>Quick Links</h3>
                        <ul className={styles.linksList}>
                            <li>
                                <Link to="/catalog" className={styles.link}>
                                    Catalog
                                </Link>
                            </li>
                            <li>
                                <Link to="#" className={styles.link}>
                                    Plant Care
                                </Link>
                            </li>
                            <li>
                                <Link to="#" className={styles.link}>
                                    Delivery & Payment
                                </Link>
                            </li>
                            <li>
                                <Link to="#" className={styles.link}>
                                    About Us
                                </Link>
                            </li>
                            <li>
                                <Link to="#" className={styles.link}>
                                    Blog
                                </Link>
                            </li>
                        </ul>
                    </div>

                    {/* ЗАМЕНИТЬ НА АКТУАЛЬНЫЕ КАТЕГОРИИ И ССЫЛКИ */}
                    <div className={styles.column}>
                        <h3 className={styles.columnTitle}>Categories</h3>
                        <ul className={styles.linksList}>
                            <li>
                                <Link to="/catalog" className={styles.link}>
                                    Large Plants
                                </Link>
                            </li>
                            <li>
                                <Link to="/catalog" className={styles.link}>
                                    Succulents
                                </Link>
                            </li>
                            <li>
                                <Link to="/catalog" className={styles.link}>
                                    Flowering Plants
                                </Link>
                            </li>
                            <li>
                                <Link to="/catalog" className={styles.link}>
                                    Hanging Plants
                                </Link>
                            </li>
                            <li>
                                <Link to="/catalog" className={styles.link}>
                                    Palms
                                </Link>
                            </li>
                        </ul>
                    </div>

                    <div className={styles.column}>
                        <h3 className={styles.columnTitle}>Contact</h3>
                        <div className={styles.contactInfo}>
                            <div className={styles.contactItem}>
                                <FaMapMarkerAlt
                                    className={styles.contactIcon}
                                />
                                <span>Paris, Rue des Jardins, 15</span>
                            </div>
                            <div className={styles.contactItem}>
                                <FaPhone className={styles.contactIcon} />
                                <span>+33 1 23 45 67 89</span>
                            </div>
                            <div className={styles.contactItem}>
                                <FaEnvelope className={styles.contactIcon} />
                                <span>info@evergreen.com</span>
                            </div>
                        </div>
                        <p className={styles.workingHours}>
                            Open daily from 9:00 to 21:00
                        </p>
                        <div className={styles.socialLinks}>
                            <Button
                                variant="outline"
                                className={styles.socialButton}
                            >
                                <FaInstagram />
                            </Button>
                            <Button
                                variant="outline"
                                className={styles.socialButton}
                            >
                                <FaFacebook />
                            </Button>
                            <Button
                                variant="outline"
                                className={styles.socialButton}
                            >
                                <FaTwitter />
                            </Button>
                        </div>
                    </div>
                </div>

                <div className={styles.copyright}>
                    <div className={styles.copyrightContent}>
                        <p className={styles.copyrightText}>
                            © {new Date().getFullYear()} Evergreen.
                        </p>
                        <p className={styles.disclaimerText}>
                            All materials on this site are used for
                            non-commercial and educational purposes only and
                            belong to their respective owners.
                        </p>
                    </div>
                </div>
            </div>
        </footer>
    );
}
