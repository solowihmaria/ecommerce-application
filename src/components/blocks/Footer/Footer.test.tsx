import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Footer } from './Footer';
import { MemoryRouter } from 'react-router-dom';

describe('<Footer/> component', () => {
    test('Render Footer component', () => {
        render(
            <MemoryRouter>
                <Footer />
            </MemoryRouter>
        );

        const footer = screen.getByTestId('footer');
        expect(footer).toBeInTheDocument();

        expect(screen.getByText('Quick Links')).toBeInTheDocument();
        expect(screen.getByText('Categories')).toBeInTheDocument();
        expect(screen.getByText('Contact')).toBeInTheDocument();
        expect(screen.getByText(/Â© \d{4} Evergreen/)).toBeInTheDocument();

        expect(footer).toMatchSnapshot();
    });

    test('Contains correct number of links', () => {
        render(
            <MemoryRouter>
                <Footer />
            </MemoryRouter>
        );

        const quickLinks = screen.getAllByTestId('quick-link');
        expect(quickLinks).toHaveLength(5);

        const categoryLinks = screen.getAllByTestId('category-link');
        expect(categoryLinks).toHaveLength(5);
    });
});
