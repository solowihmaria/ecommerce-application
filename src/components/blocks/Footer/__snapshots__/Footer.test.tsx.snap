// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> component Render Footer component 1`] = `
<footer
  class="footer"
  data-testid="footer"
>
  <div
    class="container"
  >
    <div
      class="columnsWrapper"
    >
      <div
        class="column"
      >
        <h3
          class="columnTitle"
        >
          Quick Links
        </h3>
        <ul
          class="linksList"
        >
          <li>
            <a
              class="link"
              data-discover="true"
              data-testid="quick-link"
              href="/catalog"
            >
              Catalog
            </a>
          </li>
          <li>
            <a
              class="link"
              data-discover="true"
              data-testid="quick-link"
              href="/"
            >
              Plant Care
            </a>
          </li>
          <li>
            <a
              class="link"
              data-discover="true"
              data-testid="quick-link"
              href="/"
            >
              Delivery & Payment
            </a>
          </li>
          <li>
            <a
              class="link"
              data-discover="true"
              data-testid="quick-link"
              href="/"
            >
              About Us
            </a>
          </li>
          <li>
            <a
              class="link"
              data-discover="true"
              data-testid="quick-link"
              href="/"
            >
              Blog
            </a>
          </li>
        </ul>
      </div>
      <div
        class="column"
      >
        <h3
          class="columnTitle"
        >
          Categories
        </h3>
        <ul
          class="linksList"
        >
          <li>
            <a
              class="link"
              data-discover="true"
              data-testid="category-link"
              href="/catalog"
            >
              Tropical Plants
            </a>
          </li>
          <li>
            <a
              class="link"
              data-discover="true"
              data-testid="category-link"
              href="/catalog"
            >
              Succulents
            </a>
          </li>
          <li>
            <a
              class="link"
              data-discover="true"
              data-testid="category-link"
              href="/catalog"
            >
              Flowering Plants
            </a>
          </li>
          <li>
            <a
              class="link"
              data-discover="true"
              data-testid="category-link"
              href="/catalog"
            >
              Hanging Plants
            </a>
          </li>
          <li>
            <a
              class="link"
              data-discover="true"
              data-testid="category-link"
              href="/catalog"
            >
              Ferns
            </a>
          </li>
        </ul>
      </div>
      <div
        class="column"
      >
        <h3
          class="columnTitle"
        >
          Contact
        </h3>
        <div
          class="contactInfo"
        >
          <div
            class="contactItem"
          >
            <svg
              class="contactIcon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 384 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
              />
            </svg>
            <span>
              Paris, Rue des Jardins, 15
            </span>
          </div>
          <div
            class="contactItem"
          >
            <svg
              class="contactIcon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M493.4 24.6l-104-24c-11.3-2.6-22.9 3.3-27.5 13.9l-48 112c-4.2 9.8-1.4 21.3 6.9 28l60.6 49.6c-36 76.7-98.9 140.5-177.2 177.2l-49.6-60.6c-6.8-8.3-18.2-11.1-28-6.9l-112 48C3.9 366.5-2 378.1.6 389.4l24 104C27.1 504.2 36.7 512 48 512c256.1 0 464-207.5 464-464 0-11.2-7.7-20.9-18.6-23.4z"
              />
            </svg>
            <span>
              +33 1 23 45 67 89
            </span>
          </div>
          <div
            class="contactItem"
          >
            <svg
              class="contactIcon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
              />
            </svg>
            <span>
              info@evergreen.com
            </span>
          </div>
        </div>
        <p
          class="workingHours"
        >
          Open daily from 9:00 to 21:00
        </p>
        <div
          class="socialLinks"
        >
          <button
            class="button outline socialButton"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
              />
            </svg>
          </button>
          <button
            class="button outline socialButton"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
              />
            </svg>
          </button>
          <button
            class="button outline socialButton"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      class="copyright"
    >
      <div
        class="copyrightContent"
      >
        <p
          class="copyrightText"
        >
          Â© 
          2025
           Evergreen.
        </p>
        <p
          class="disclaimerText"
        >
          All materials on this site are used for non-commercial and educational purposes only and belong to their respective owners.
        </p>
      </div>
    </div>
  </div>
</footer>
`;
